### ISSUE TITLE
CSRF Meta tag not getting updated.

### ISSUE THREAD

**Issue by darylthornhill** [GeneaLabs/laravel-caffeine#90]
## Expected Behavior
csrf-token meta should get updated on drip.

## Acutal Behavior
Im using axios, I am prepending the csrf-token from the meta to the headers of the axios request.
If i manually update my csrf meta tag then it gets picked up in axios however drip doesn't update the csrf-token meta.

## Environment
- PHP Version: 7.1
- Laravel Version: 5.6
- LaravelCaffeine Version: ^0.6.8

## Stack Trace
Standard token mismatch error.

**Comment by ItaloBC** [comment]
Laravel Caffeine isn't compatible with SPA sites. POST methods will fail nonetheless whataver you are using,

There are two ways to tackle the problem:

- Use JWT for your forms.
- [Refresh the CSRF Token](https://medium.com/@serhii.matrunchyk/laravel-vue-spas-how-to-send-ajax-requests-and-not-run-into-csrf-tokens-mismatch-exceptions-da3b71b287ab) using Axios (or equivalent) to catch de 429 code that Laravel threw up.

**Comment by darylthornhill** [comment]
@ItaloBC Surely if drip is supposed to update the meta tag of the html, then i can just grab that meta tag when i post though?

Currently Drip doesn't update the meta tag, it isn't changing. Even on standard blade pages. 

The drip is running in the network tab in the developer tools window too.

**Comment by ItaloBC** [comment]
@darylthornhill  No. The drip doens't update the token.

Waht Laravel Caffeine does is poking your App with a GET, which returns a [204 code](https://httpstatuses.com/204) (response OK with no body).

Before returning the response, in layman's terms, Laravel checks that the cookie_session is sent with the drip matches the session entry (a file, a database row, or whatever your driver is). If it is true, it will renew the token expiration for the time configured time in your app (120 min default).

If not, the token will be marked as expired, and it will throw a 419 code with TokenMismatchException after the form (with the token) is sent.

**Comment by darylthornhill** [comment]
Okay that makes sense why the meta isn’t changing.

However it doesn’t really make sense why my token is expiring.

If I set my token to expire in 4 mins and drip every 10 secs. After 4 mins
I get a token mismatch error. If using axios or another Ajaxy method.

**Comment by ItaloBC** [comment]
@darylthornhill May be it's related to #89 

In my app my forms are expiring, and it's not SPA, regardless of the configuration and dripping status.

You can set your session lifetime to 1 min, and change this in `genealabs-laravel-caffeine.php`:

```php
'drip-interval' => 20000,
'outdated-drip-check-interval' => 0,
``` 

Make a view with a simple form and submit it after the minute. If everything is set correctly, you should get a successful submission. At least, for me, it doesn't work unless I add manually the dripping route to my `web.php`.

**Comment by darylthornhill** [comment]
I just gave it a go and set the route up in my web.php routes file and it works. I'm pretty sure because in the web.php routes its hitting the verify csrf middleware. 

As far as i can see the caffeine drip doesnt hit that middleware. That could be the problem?

**Comment by ItaloBC** [comment]
@darylthornhill No, the dripping isn't related to verifying the CSRF Token. Otherwise, making a GET to any other route shouldn't refresh the session either.

**Comment by darylthornhill** [comment]
Nah I meant the verify csrf middleware is the place that looks to refresh
the xsrf cookie.

Does drip rely on that?

Hence why creating your own route in the web.php works because it refreshes
that xsrf token and when it’s posted it’ll be a valid non expired token.

**Comment by darylthornhill** [comment]
I can see that caffeine looks for the web middleware.

The only reason why I can see that me manually adding the route vs using
just the caffeine plugin works is that maybe caffeine can’t see that the
web middleware exists.

I’ll have a gander into the source code of the plugin tomorrow and let you
know.

**Comment by ItaloBC** [comment]
You are right, caffeine doesn't sees the 'web' middleware-group. I think I know how it can be fixed: by manually adding the route with the middlware-group

**Comment by mikebronner** [comment]
I'll close this as the csrf-token is not meant to be updated. Looks like you already started the other issue. :)
